server/container/guice/postgres-common/src/main/java/org/apache/james/modules/data/PostgresCommonModule.java

```diff
@@ -102,16 +103,29 @@ PostgresModule composePostgresDataDefinitions(Set<PostgresModule> modules) {
 
     @Provides
     @Singleton
-    PostgresTableManager postgresTableManager(PostgresExecutor.Factory factory,
+    PostgresTableManager postgresTableManager(PostgresExecutor postgresExecutor,
                                               PostgresModule postgresModule,
                                               PostgresConfiguration postgresConfiguration) {
-        return new PostgresTableManager(factory, postgresModule, postgresConfiguration);
+        return new PostgresTableManager(postgresExecutor, postgresModule, postgresConfiguration);
     }
 
     @Provides
     @Named(DEFAULT_INJECT)
     @Singleton
-    PostgresExecutor defaultPostgresExecutor(PostgresTableManager postgresTableManager) {
-        return postgresTableManager.get();
+    PostgresExecutor defaultPostgresExecutor(PostgresExecutor.Factory factory) {
+        return factory.create();
+    }
+
+    @Provides
+    @Singleton
+    PostgresExecutor postgresExecutor(@Named(DEFAULT_INJECT) PostgresExecutor postgresExecutor) {
+        return postgresExecutor;
+    }
+
+    @ProvidesIntoSet
+    InitializationOperation provisionPostgresTablesAndIndexes(PostgresTableManager postgresTableManager) {
+        return InitilizationOperationBuilder
+            .forClass(PostgresTableManager.class, 2)
```

Comment:
Maybe a constant to avoid the magic number? Like `HIGH_PRIORITY`. Or whatever better name.